x-pack/plugins/apm/public/hooks/use_transaction_latency_chart_fetcher.ts

```diff
@@ -22,15 +23,30 @@ export function useTransactionLatencyChartsFetcher() {
     urlParams: { start, end, transactionName },
     uiFilters,
   } = useUrlParams();
+  const { environments } = useEnvironmentsFetcher({ start, end, serviceName });
```

Comment:
This might lead to duplicate loading of the environments since `useEnvironmentsFetcher` is already called in the `EnvironmentFilter`. 
I originally wanted to add environment to the `ApmServiceContext` but it was only used in a single spot (apart from alerting flyouts). Seeing it's now used several places it might make sense to lift it up to `ApmServiceContext` and consume it from there.